drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/DriftNettyMethodInvokerFactory.java

```diff
@@ -51,8 +55,10 @@
     private final EventLoopGroup group;
     private final SslContextFactory sslContextFactory;
     private final Optional<HostAndPort> defaultSocksProxy;
-    private final ConnectionManager connectionManager;
+    private final ConnectionFactory connectionFactory;
     private final ScheduledExecutorService connectionPoolMaintenanceExecutor;
+    private final DriftNettyConnectionFactoryConfig factoryConfig;
+    private final Map<Optional<I>, ConnectionPool> connectionPools = new ConcurrentHashMap<>();
```

Comment:
So client identity is always hashable, right?

Reply:
Yup. In production it is always `DriftClientAnnotationImpl `: https://github.com/prestodb/drift/blob/master/drift-client/src/main/java/com/facebook/drift/client/guice/DriftClientAnnotationFactory.java#L65